src/main/java/org/apache/commons/imaging/ImageParser.java

```diff
@@ -682,8 +688,10 @@ public abstract Dimension getImageSize(ByteSource byteSource, Map<String, Object
      *                            parser implementation.
      * @throws IOException        In the event of unsuccessful read or access operation.
      */
-    public abstract String getXmpXml(ByteSource byteSource, Map<String, Object> params)
-            throws ImageReadException, IOException;
+    public String getXmpXml(final ByteSource byteSource, final Map<String, Object> params)
```

Comment:
My feeling is, that we have a general design flaw here. If we have a a super class A and to inheriting class B and C, it makes no sense to define methods in A which are only supported by either B oder C but not both. If a method is added to A that doesn't make sense for either B or C then A doesn't seem to be the generalization of B and C.